.devcontainer/adx/create-adx.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+
+# Display Help
+Help()
+{
+   echo "Create data explorer cluster for development purpose."
+   echo
+   echo "Syntax: create-adx.sh [-u|h]"
+   echo "options:"
+   echo "u     Unique name used to create resources."
+   echo "h     Print this Help."
+   echo
+}
+
+while getopts u:h flag;
+do
+    case "${flag}" in
+        h) Help
+           exit;;
+        u) unique=${OPTARG};;
+    esac
+done
+
+if [ ! "$unique" ]; then
+    echo "Please add unique name argument."
+    exit;
+fi
+
+# Resource Names
+RESOURCE_GROUP_LOCATION="westeurope"
+RESOURCE_GROUP_NAME="rg-k2-$unique-dev"
+ADX_CLUSTER_NAME="adxk2$unique"
+ADX_DB_NAME="dev"
+STORAGE_ACCOUNT_NAME="stk2$unique"
+STORAGE_CONTAINER_NAME="scripts"
+SERVICE_PRINCIPAL_NAME="sp-k2-$unique"
+
+# Create Resource Group
+RESOURCE_GROUP_SCOPE=$(az group create --location $RESOURCE_GROUP_LOCATION --resource-group $RESOURCE_GROUP_NAME --query id -o tsv)
+
+# Create Data Explorer Cluster
+az extension add -n kusto
+echo "$(tput setaf 2)[1/8] Creating Data Explorer Cluster. Please be patient. Long running operation (10 minutes).$(tput setaf 7)"
+az kusto cluster create \
+    --cluster-name $ADX_CLUSTER_NAME \
+    --sku name="Dev(No SLA)_Standard_D11_v2" tier="Basic" capacity=1 \
+    --resource-group $RESOURCE_GROUP_NAME \
+    --location $RESOURCE_GROUP_LOCATION \
+    --enable-streaming-ingest true \
+    --type="None"
+
+# Create Data Explorer Database
+echo "$(tput setaf 2)[2/8] Creating Data Explorer Database.$(tput setaf 7)"
+az kusto database create \
+    --cluster-name $ADX_CLUSTER_NAME \
+    --database-name $ADX_DB_NAME \
+    --resource-group $RESOURCE_GROUP_NAME \
+    --read-write-database soft-delete-period=P365D hot-cache-period=P31D location=$RESOURCE_GROUP_LOCATION
+
+# Create Service Principal
+echo "$(tput setaf 2)[3/8] Creating Service Principal.$(tput setaf 7)"
+SECRETS=$(az ad sp create-for-rbac --name $SERVICE_PRINCIPAL_NAME --role Contributor --scopes $RESOURCE_GROUP_SCOPE)
+SERVICE_PRINCIPAL_FQN="aadapp=$(echo $SECRETS | jq -r '.appId');$(echo $SECRETS | jq -r '.tenant')"
+SERVICE_PRINCIPAL_DISPLAYNAME=$(echo $SECRETS | jq '.displayName')
+
+# Wait Service Principal propagation
+echo "$(tput setaf 2)[4/8] Wait 60 seconds until Service Principal is propagated.$(tput setaf 7)"
+sleep 60
+
+# Assign Service Principal to database permissions
+echo "$(tput setaf 2)[5/8] Assigning database permissions.$(tput setaf 7)"
```

Comment:
I would output the SP id, so the user knows which SP to delete when done with the env.
(also can print a reminder message about deleting it)

Reply:
Yes. SP id was already printed at the end of the script; but I added a reminder message.

At the end of your development, don't forget to delete service principal and its role assignements.
az ad sp delete --id b4edb1f6-e538-4071-93ea-xxxxxxxxxxxx